spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/SparkTestBaseWithCatalog.java

```diff
@@ -44,9 +45,7 @@ public static void createWarehouse() throws IOException {
 
   @AfterClass
   public static void dropWarehouse() {
-    if (warehouse != null && warehouse.exists()) {
-      warehouse.delete();
-    }
+    FileUtils.deleteQuietly(warehouse);
```

Comment:
This might hide unexpected exceptions which can indicate problems. I have preferred the original solution

Reply:
Do you mean we should not suppress exceptions if the folder can't be deleted? Then I guess we can use `FileUtils::forceDelete`, which has the following in java doc:
```java
     * Deletes a file. If file is a directory, delete it and all sub-directories.
     * <p>
     * The difference between File.delete() and this method are:
     * <ul>
     * <li>A directory to be deleted does not have to be empty.</li>
     * <li>You get exceptions when a file or directory cannot be deleted.
     *      (java.io.File methods returns a boolean)</li>
     * </ul>
```
The problem with the original solution is it can't delete this folder.

Reply:
Using `Assert` here as you do in the new PR versions looks good to me